Vault/Library/CachePersistence.cs

```diff
@@ -124,10 +128,93 @@ void AfterAccessNotification(AuthenticationRecord authRecord)
                 // Call AuthenticateAsync to fetch a new AuthenticationRecord.
                 authRecord = credential.Authenticate();
 
+                // Create the original data to be encrypted
+                var toEncrypt = Encoding.ASCII.GetBytes(authRecord.ToString());

```

Comment:
Could lead to data corruption and transcoding issues. Should use `Encoding.UTF8`.